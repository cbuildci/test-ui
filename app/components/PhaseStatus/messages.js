/*
 * PhaseStatus Messages
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    phaseStatus: {
        id: 'app.components.PhaseStatus.phaseStatus',
        defaultMessage: `{status, select,
            other {{status}}
        }`,
    },
});
