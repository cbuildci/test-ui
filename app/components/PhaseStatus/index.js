/**
 * PhaseStatus
 */

import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

export const statusDisplay = {
    SUCCEEDED: { icon: 'fas fa-check', color: '#28a745' },
    FAILED: { icon: 'fas fa-times', color: '#dc3545' },
    FAULT: { icon: 'fas fa-times', color: '#dc3545' },
    TIMED_OUT: { icon: 'far fa-clock', color: '#dc3545' },
    IN_PROGRESS: { icon: 'fas fa-hourglass-start', color: '#17a2b8' },
    STOPPED: { icon: 'fas fa-stop-circle', color: '#dc3545' },
};

function PhaseStatus({ status }) {
    const { icon, color } = statusDisplay[status] || {};

    return (
        <span style={{ color }}>
            {icon && <i className={`${icon} fa-fw`}/>}
            {icon && ' '}
            {status ? <FormattedMessage
                {...messages.phaseStatus}
                values={{ status: status }}
            /> : '-'}
        </span>
    );
}

PhaseStatus.propTypes = {
    status: PropTypes.string,
};

PhaseStatus.defaultProps = {
    status: null,
};

export default PhaseStatus;
