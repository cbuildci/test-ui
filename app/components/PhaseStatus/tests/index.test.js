import React from 'react';
import { shallow } from 'enzyme';
import PhaseStatus, { statusDisplay } from '../index';

describe('<PhaseStatus />', () => {
    it('should have expected statuses for statusDisplay', () => {
        expect(Object.keys(statusDisplay).sort())
            .toEqual([
                'SUCCEEDED',
                'FAILED',
                'FAULT',
                'TIMED_OUT',
                'IN_PROGRESS',
                'STOPPED',
            ].sort());
    });

    it('should render expected JSX for statuses', () => {
        for (const status of Object.keys(statusDisplay)) {
            expect(shallow(
                <PhaseStatus status={status}/>,
            )).toMatchSnapshot();
        }
    });

    it('should render expected JSX for unknown status', () => {
        expect(shallow(
            <PhaseStatus status="foobar"/>,
        )).toMatchSnapshot();

        expect(shallow(
            <PhaseStatus/>,
        )).toMatchSnapshot();
    });

});
