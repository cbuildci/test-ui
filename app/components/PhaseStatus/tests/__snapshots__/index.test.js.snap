// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhaseStatus /> should render expected JSX for status "" 1`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "" 2`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "FAILED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "FAILED",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "FAILED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    FAILED
  </span>
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "FAULT" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "FAULT",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "FAULT" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    FAULT
  </span>
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "IN_PROGRESS" 1`] = `
<span
  style={
    Object {
      "color": "#17a2b8",
    }
  }
>
  <i
    className="fas fa-hourglass-start fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "IN_PROGRESS",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "IN_PROGRESS" 2`] = `
<span
  style="color:#17a2b8"
>
  <i
    class="fas fa-hourglass-start fa-fw mr-1"
  />
  <span>
    IN_PROGRESS
  </span>
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "STOPPED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-stop-circle fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "STOPPED",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "STOPPED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-stop-circle fa-fw mr-1"
  />
  <span>
    STOPPED
  </span>
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "SUCCEEDED" 1`] = `
<span
  style={
    Object {
      "color": "#28a745",
    }
  }
>
  <i
    className="fas fa-check fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "SUCCEEDED",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "SUCCEEDED" 2`] = `
<span
  style="color:#28a745"
>
  <i
    class="fas fa-check fa-fw mr-1"
  />
  <span>
    SUCCEEDED
  </span>
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "TIMED_OUT" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="far fa-clock fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "TIMED_OUT",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for status "TIMED_OUT" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="far fa-clock fa-fw mr-1"
  />
  <span>
    TIMED_OUT
  </span>
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for statuses 1`] = `
<span
  style={
    Object {
      "color": "#28a745",
    }
  }
>
  <i
    className="fas fa-check fa-fw"
  />
   
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "SUCCEEDED",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for statuses 2`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw"
  />
   
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "FAILED",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for statuses 3`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw"
  />
   
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "FAULT",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for statuses 4`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="far fa-clock fa-fw"
  />
   
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "TIMED_OUT",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for statuses 5`] = `
<span
  style={
    Object {
      "color": "#17a2b8",
    }
  }
>
  <i
    className="fas fa-hourglass-start fa-fw"
  />
   
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "IN_PROGRESS",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for statuses 6`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-stop-circle fa-fw"
  />
   
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "STOPPED",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for unknown status 1`] = `
<span
  style={
    Object {
      "color": undefined,
    }
  }
>
  <FormattedMessage
    defaultMessage="{status, select,
            other {{status}}
        }"
    id="app.components.PhaseStatus.phaseStatus"
    values={
      Object {
        "status": "foobar",
      }
    }
  />
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for unknown status 2`] = `
<span
  style={
    Object {
      "color": undefined,
    }
  }
>
  -
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for unknown status 3`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<PhaseStatus /> should render expected JSX for unknown status 4`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;
